dulwich/object_store.py

```diff
@@ -557,14 +560,19 @@ def _iter_loose_objects(self):
             if len(base) != 2:
                 continue
             for rest in os.listdir(os.path.join(self.path, base)):
-                yield (base+rest).encode(sys.getfilesystemencoding())
+                fpath = (base+rest).encode(sys.getfilesystemencoding())
+                if not self._get_loose_object(fpath):
+                    continue
+                yield fpath
 
     def _get_loose_object(self, sha):
         path = self._get_shafile_path(sha)
         try:
-            if os.lstat(path).st_size == 0:
-                return None
             return ShaFile.from_path(path)
+        except CorruptedFormatException:
```

Comment:
You'd probably want to let this bubble up to the caller. If they explicitly asked for this object (rather than e.g. as part of iterating all the objects in the repo), they'd want to know about corruption.